src/cast_core/include/cast_core/actors/OptionsConversion.hpp

```diff
@@ -26,216 +26,200 @@
+            auto bypassDocValidation = node["BypassDocumentValidation"].to<bool>();
             rhs.bypass_document_validation(bypassDocValidation);
         }
         if (node["Ordered"]) {
-            auto isOrdered = node["Ordered"].as<bool>();
+            auto isOrdered = node["Ordered"].to<bool>();
             rhs.ordered(isOrdered);
         }
         if (node["WriteConcern"]) {
-            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            auto wc = node["WriteConcern"].to<mongocxx::write_concern>();
             rhs.write_concern(wc);
         }
-        return true;
+        return rhs;
     }
 };
 
 template <>
-struct convert<mongocxx::options::count> {
-    using CountOptions = mongocxx::options::count;
-    static Node encode(const CountOptions& rhs) {
-        return {};
-    }
+struct NodeConvert<mongocxx::options::count> {
+    using type = mongocxx::options::count;
+
+    static type convert(const Node& node) {
+        type rhs{};
 
-    static bool decode(const Node& node, CountOptions& rhs) {
-        if (!node.IsMap()) {
-            return false;
-        }
         if (node["Hint"]) {
-            auto h = node["Hint"].as<std::string>();
-            auto hint = mongocxx::hint(h);
+            auto h = node["Hint"].to<std::string>();
+            auto hint = mongocxx::hint(std::move(h));
             rhs.hint(hint);
         }
         if (node["Limit"]) {
-            auto limit = node["Limit"].as<int>();
+            auto limit = node["Limit"].to<int>();
             rhs.limit(limit);
         }
         if (node["MaxTime"]) {
-            auto maxTime = node["MaxTime"].as<genny::TimeSpec>();
+            auto maxTime = node["MaxTime"].to<genny::TimeSpec>();
             rhs.max_time(std::chrono::milliseconds{maxTime});
         }
         if (node["ReadPreference"]) {
-            auto readPref = node["ReadPreference"].as<mongocxx::read_preference>();
+            auto readPref = node["ReadPreference"].to<mongocxx::read_preference>();
             rhs.read_preference(readPref);
         }
-        return true;
+        return rhs;
     }
 };
 
 template <>
-struct convert<mongocxx::options::find> {
-    using FindOptions = mongocxx::options::find;
-    static Node encode(const FindOptions& rhs) {
-        return {};
-    }
+struct NodeConvert<mongocxx::options::find> {
+    using type = mongocxx::options::find;
+
+    static type convert(const Node& node) {
+        type rhs{};
 
-    static bool decode(const Node& node, FindOptions& rhs) {
-        if (!node.IsMap()) {
-            return false;
-        }
         if (node["Hint"]) {
-            auto h = node["Hint"].as<std::string>();
-            auto hint = mongocxx::hint(h);
-            rhs.hint(hint);
+            auto h = node["Hint"].to<std::string>();
+            auto hint = mongocxx::hint(std::move(h));
+            rhs.hint(mongocxx::hint(hint));
```

Comment:
TIG-2937 new options (and the std::move seems to be required or the string goes out of scope).